akka-actor/src/main/scala/akka/event/Logging.scala

```diff
@@ -1627,17 +1627,36 @@ trait DiagnosticLoggingAdapter extends LoggingAdapter {
 
 /** DO NOT INHERIT: Class is open only for use by akka-slf4j*/
 @DoNotInherit
-class LogMarker(val name: String)
+class LogMarker(val name: String, val properties: Map[String, Any]) {
+
+  // for binary compatibility
+  def this(name: String) = this(name, Map.empty)
+
+  /** Java API */
+  def getProperties: java.util.Map[String, Object] = {
+    import akka.util.ccompat.JavaConverters._
+    properties.map { case (k, v) => (k, v.asInstanceOf[AnyRef]) }.asJava
```

Comment:
Could we type that map to contain `AnyRef`s, since that's effectively what will happen anyway? or does that lead to a cumbersome API elsewhere?

Reply:
I tried and some of our usages doesn't like that:
```
[error] /Users/patrik/dev/akka/akka-remote/src/main/scala/akka/remote/RemoteLogMarker.scala:39:76: type mismatch;
[error]  found   : Any
[error]  required: AnyRef
[error]         LogMarker.Properties.RemoteAddressUid -> remoteAddressUid.getOrElse("")))
```

I don't expect this Java API to be used much, in fact I don't think applications should use this at all. They should use plain SL4F or Logstash APIs when logging things like this. We need the API to be public so that we can use it in other projects, such as Akka Management.